docs/deploy/services/kubernetes.md

```diff
@@ -23,9 +23,45 @@ If your services are running over HTTP2 (the default), each Restate partition wi
 However, because there are many partitions (by default 24, typically more in a larger cluster) your pods should get a reasonably
 even distribution of traffic even without a L7 load balancing solution (Cilium, Istio etc).
 
+### RestateDeployment CRD
+The [Restate operator](https://github.com/restatedev/restate-operator) allows for the use of a `RestateDeployment` CRD to deploy your services.
+The CRD is an extension of a native `Deployment` object, but will manage registration and [versioning](/operate/versioning) for you, by keeping old
+ReplicaSets around with an associated Service object so that in-flight invocations can drain against the old code versions. You can deploy a CRD as follows:
+
+```yaml
+apiVersion: restate.dev/v1beta1
+kind: RestateDeployment
+metadata:
+  name: service
+spec:
+  replicas: 1
+  restate:
+    register:
+      # set this if you want to register against your RestateCluster named 'restate'
+      # alternatively, you can set a url or a Service reference to register against
+      cluster: restate
+  selector:
+    matchLabels:
+      app: service
+  template:
+    metadata:
+      labels:
+        app: service
+    spec:
+      containers:
+        - name: service
+          image: path.to/yourrepo:yourtag
+          env:
+            - name: PORT
+              value: "9080"
+          ports:
+            - containerPort: 9080
+              name: restate
```

Comment:
If the user doesn't specify this, perhaps we can inject this ourselves?

Reply:
the operator defaults to 9080 anyway, is it necessary? i want to treat their pod spec as opaque as possible ideally